azure/table_azure_compute_virtual_machine_size.go

```diff
@@ -0,0 +1,134 @@
+			},
+			{
+				Name:        "number_of_cores",
+				Description: "The number of cores supported by the virtual machine size.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("VirtualMachineSize.NumberOfCores"),
+			},
+			{
+				Name:        "os_disk_size_in_mb",
+				Description: "The OS disk size, in MB, allowed by the virtual machine size.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("VirtualMachineSize.OSDiskSizeInMB"),
+			},
+			{
+				Name:        "resource_disk_size_in_mb",
+				Description: "The resource disk size, in MB, allowed by the virtual machine size.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("VirtualMachineSize.ResourceDiskSizeInMB"),
+			},
+			{
+				Name:        "memory_in_mb",
+				Description: "The amount of memory, in MB, supported by the virtual machine size.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("VirtualMachineSize.MemoryInMB"),
+			},
+			{
+				Name:        "max_data_disk_count",
+				Description: "The maximum number of data disks that can be attached to the virtual machine size.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("VirtualMachineSize.MaxDataDiskCount"),
+			},
+
+			// Standard steampipe columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("VirtualMachineSize.Name"),
+			},
+
+			// Azure standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Location").Transform(toLower),
+			},
+		}),
+	}
+}
+
+type VMSizeInfo struct {
+	*armcompute.VirtualMachineSize
+	Location string
+}
+
+//// LIST FUNCTION ////
+
+func listComputeVirtualMachineSizes(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	if h.Item == nil {
+		return nil, nil
+	}
+
+	locationDetails := h.Item.(sub.Location)
+
+	session, err := GetNewSessionUpdated(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_compute_virtual_machine_size.listComputeVirtualMachineSizes", "session_error", err)
+		return nil, err
+	}
+
+	session.ClientOptions.APIVersion = "2024-07-01"
+
+	clientFactory, err := armcompute.NewVirtualMachineSizesClient(session.SubscriptionID, session.Cred, session.ClientOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	pager := clientFactory.NewListPager(*locationDetails.Name, &armcompute.VirtualMachineSizesClientListOptions{})
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			if strings.Contains(strings.ToLower(err.Error()), "no registered resource provider found for location") {
+				plugin.Logger(ctx).Error("azure_compute_virtual_machine_size.listComputeVirtualMachineSizes", "no registered resource provider found", err.Error())
+				return nil, nil
```

Comment:
@ParthaI why are we handling this error? 

When does the API return this error?